src/components/NcRadioGroup/NcRadioGroup.vue

```diff
@@ -0,0 +1,198 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
+<script setup lang="ts">
+import type { Slot } from 'vue'
+
+import { computed, provide } from 'vue'
+import { INSIDE_RADIO_GROUP_KEY } from './useNcRadioGroup.ts'
+
+const modelValue = defineModel<string>({ required: false, default: '' })
+
+defineProps<{
+	/**
+	 * If the radio group should be shown as buttons.
+	 * When set `NcRadioGroupButton` must be used in the default slot.
+	 */
+	buttonVariant?: boolean
```

Comment:
Can't it be detected based on the content of the slot?

Reply:
we can do this yes - more by using inject / provide instead of parsing the slot content (we messed this already up in actions :sweat_smile: ).
What do you think @ShGKme ?